pkg/plugin/plugin_test.go

```diff
@@ -20,211 +21,52 @@ func TestCreateOrUpdate(t *testing.T) {
-			isUpdate: true,
-			wantErr:  true,
-			errStep:  api.PluginStepUpdateMasterAgentPoolListVMs,
-		},
-		{
-			name:     "update in place: read blob error",
-			isUpdate: true,
-			wantErr:  true,
-			errStep:  api.PluginStepUpdateMasterAgentPoolReadBlob,
-		},
-		{
-			name:     "update in place: drain error",
-			isUpdate: true,
-			wantErr:  true,
-			errStep:  api.PluginStepUpdateMasterAgentPoolDrain,
-		},
-		{
-			name:     "update in place: deallocate error",
-			isUpdate: true,
-			wantErr:  true,
-			errStep:  api.PluginStepUpdateMasterAgentPoolDeallocate,
-		},
-		{
-			name:     "update in place: update VMs error",
-			isUpdate: true,
-			wantErr:  true,
-			errStep:  api.PluginStepUpdateMasterAgentPoolUpdateVMs,
-		},
-		{
-			name:     "update in place: reimage error",
-			isUpdate: true,
-			wantErr:  true,
-			errStep:  api.PluginStepUpdateMasterAgentPoolReimage,
-		},
-		{
-			name:     "update in place: start error",
-			isUpdate: true,
-			wantErr:  true,
-			errStep:  api.PluginStepUpdateMasterAgentPoolStart,
-		},
-		{
-			name:     "update in place: wait for ready error",
-			isUpdate: true,
-			wantErr:  true,
-			errStep:  api.PluginStepUpdateMasterAgentPoolWaitForReady,
-		},
-		{
-			name:     "update in place: update blob error",
-			isUpdate: true,
-			wantErr:  true,
-			errStep:  api.PluginStepUpdateMasterAgentPoolUpdateBlob,
-		},
-		{
-			name:     "update plus one: list VMs error",
-			isUpdate: true,
-			wantErr:  true,
-			errStep:  api.PluginStepUpdateWorkerAgentPoolListVMs,
-		},
-		{
-			name:     "update plus one: read blob error",
-			isUpdate: true,
-			wantErr:  true,
-			errStep:  api.PluginStepUpdateWorkerAgentPoolReadBlob,
-		},
-		{
-			name:     "update plus one: wait for ready error",
-			isUpdate: true,
-			wantErr:  true,
-			errStep:  api.PluginStepUpdateWorkerAgentPoolWaitForReady,
-		},
-		{
-			name:     "update plus one: update blob error",
-			isUpdate: true,
-			wantErr:  true,
-			errStep:  api.PluginStepUpdateWorkerAgentPoolUpdateBlob,
-		},
-		{
-			name:     "waitforinfra: daemon error",
-			isUpdate: true,
-			wantErr:  true,
-			errStep:  api.PluginStepWaitForInfraDaemonSets,
-		},
-		{
-			name:     "waitforinfra: deployments error",
-			isUpdate: true,
-			wantErr:  true,
-			errStep:  api.PluginStepWaitForInfraDeployments,
-		},
-		{
-			name:     "ConsoleHealth: error",
-			isUpdate: true,
-			wantErr:  true,
-			errStep:  api.PluginStepWaitForConsoleHealth,
+	}
+	deployer := func(ctx context.Context, azuretemplate map[string]interface{}) error {
+		return nil
+	}
+	cs := &api.OpenShiftManagedCluster{
+		Properties: api.Properties{
+			AgentPoolProfiles: []api.AgentPoolProfile{
+				{Role: api.AgentPoolProfileRoleMaster, Name: "master"},
+				{Role: api.AgentPoolProfileRoleCompute, Name: "compute"},
+				{Role: api.AgentPoolProfileRoleInfra, Name: "infra"},
+			},
 		},
 	}
+
 	for _, tt := range tests {
 		mockGen := mock_arm.NewMockGenerator(mockCtrl)
```

Comment:
it's a complete nit but I wonder if it's worth trying to standardise in the direction of calling the mocks by the fieldname they go into, ie s/mockGen/armGenerator/, s/mockUp/clusterUpgrader/. See update_master_test.go as an example.

Reply:
done